Updated salt/modules/disk.py to fix postPrefixes by a factor of 10. p…

vib795 opened the pull request:
### What does this PR do?

### What issues does this PR fix or reference?
Fixes: #65490

### Previous Behavior
1.0K gives 10000.0

### New Behavior
1.0K should give 1000.0

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**
<!-- Please review the [test documentation](https://docs.saltproject.io/en/master/topics/tutorials/writing_tests.html) for details on how to implement tests into Salt's test suite. -->
- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes/No

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

s0undt3ch commented:
We'll need unit tests around `_parse_numbers` please.

leeclemens commented:
@vib795 IMHO, just a minor thought, but it may be more symmetrical and slightly easier to read if they are formatted the                 same way `decimal.Decimal` displays the values (e.g. `1E+12`).

twangboy reviewed (changes requested):
This needs a changelog. Please migrate the test to pytests.

dwoz committed:
Fix disk._parse_numbers SI suffix multipliers off by 10x

postPrefixes used "10E3" (=10000) instead of "1E3" (=1000), causing
all values with K/M/G/T/P/E/Z/Y suffixes to be inflated by a factor
of 10. Fix multipliers and add pytest parametrized tests.

Fixes: #65490
